src/esn.py

```diff
@@ -29,16 +34,17 @@ def __init__(self, config: ESNConfig, prng: Optional[Any]) -> None:
         self.spectral_radius = config.spectral_radius
         self.feedback = config.feedback
         self.skip_connections = config.skip_connections
+        self.architecture = config.architecture
+        self.top_layer = config.top_layer
+        self.random_feature = config.random_feature
+        self.random_feature_size = config.random_feature_size
 
         # PRNG key
-        if prng is None:
-            self.prng = np.random.default_rng()
-        else:
-            assert isinstance(prng, np.random.Generator)
-            self.prng = prng
+        assert isinstance(prng, np.random.Generator)
+        self.prng = prng
 
         # shortcut
-        K, N, L = self.get_sizes()
+        K, N, N_rfc, L = self.get_sizes()
```

Comment:
Just for consistency, we could call this new size **F**?

Reply:
M is okay? f is used for a matrix in RFC

Reply:
👍